src/emulate.c

```diff
@@ -958,6 +955,47 @@ static bool rv_has_plic_trap(riscv_t *rv)
     return ((rv->csr_sstatus & SSTATUS_SIE || !rv->priv_mode) &&
             (rv->csr_sip & rv->csr_sie));
 }
+
+static void rv_check_interrupt(riscv_t *rv)
+{
+    vm_attr_t *attr = PRIV(rv);
+    /* check for any interrupt after every block/instruction emulation */
```

Comment:
Move this comment to the beginning of the while loop in the `rv_step` since `rv_step_debug` call it step by step.